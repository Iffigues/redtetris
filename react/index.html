<!doctype html>
<html>
  <head>
    <title>RedTetris</title>
    <link rel="icon" href="http://redpelicans.com/favicon.ico">                                                                                                                                          
  </head>
  <body style="margin: 0">
    <div id="tetris"></div>
    <script src="/bundle.js"></script>
  </body>
</html>
