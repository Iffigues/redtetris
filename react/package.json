{
  "name": "red_tetrisboilerplate",
  "version": "0.0.1",
  "author": "redpelicans",
  "license": "MIT",
  "scripts": {
    "eslint": "eslint src",
    "client-dist": "NODE_ENV=production webpack --progress --colors",
    "client-dev": "webpack-dev-server --colors --hot --inline --host 0.0.0.0 --port 8080",
    "srv-dev": "DEBUG=tetris:* babel-watch -w src src/server/main.js",
    "srv-dist": "DEBUG=tetris:* babel src --out-dir dist",
    "test": "mocha --require babel-core/register --reporter spec",
    "coverage": "NODE_ENV=test nyc --include src --all -r text -r lcov -r text --require babel-core/register mocha "
  },
  "dependencies": {
    "@babel/preset-es2017": "^7.0.0-beta.53",
    "@babel/preset-react": "^7.10.4",
    "eslint-config-airbnb": "^18.2.0",
    "react": "^15.0.2",
    "react-dom": "^15.0.2",
    "react-hot-loader": "^4.12.21",
    "react-redux": "^4.4.5",
    "redux": "^3.5.2",
    "redux-logger": "^2.6.1",
    "redux-thunk": "^2.0.1",
    "socket.io": "^2.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.10.5",
    "@babel/preset-env": "^7.10.4",
    "babel-cli": "^6.7.7",
    "babel-core": "^6.7.7",
    "babel-eslint": "^6.1.0",
    "babel-loader": "^8.1.0",
    "babel-plugin-__coverage__": "^0.111111.1",
    "babel-preset-es2015": "^6.6.0",
    "babel-preset-react": "^6.5.0",
    "babel-preset-stage-0": "^6.5.0",
    "babel-watch": "^2.0.2",
    "chai": "^3.5.0",
    "chai-as-promised": "^5.3.0",
    "chai-equal-jsx": "^1.0.9",
    "eslint": "^4.18.2",
    "eslint-plugin-babel": "^3.3.0",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-react": "^5.2.2",
    "mocha": "^8.0.1",
    "nyc": "^15.1.0",
    "webpack": "^4.43.0",
    "webpack-cli": "^3.3.12",
    "webpack-dev-middleware": "^1.6.1",
    "webpack-dev-server": "^3.11.0",
    "webpack-hot-middleware": "^2.10.0"
  },
  "nyc": {
    "exclude": [
      "build"
    ]
  }
}
