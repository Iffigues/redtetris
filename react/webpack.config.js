const path = require('path');

module.exports = {
  entry: {
    app: './src/client/index.js'
  },
  output: {
    filename: 'bundle.js',
    path: path.resolve(__dirname, 'dist')
  },
  module: {
    rules: [
      {
        test: /\.(js|jsx)$/,
        exclude: /node_modules/,
        use: [{
          loader: 'babel-loader',
          options: {
              cacheDirectory: true,
              babelrc: false,
              presets: [["@babel/env"], "@babel/react"],
              plugins: ["react-hot-loader/babel"]
          }
        }]
      }
    ]
  }
};
