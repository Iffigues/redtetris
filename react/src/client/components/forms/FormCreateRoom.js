import React from 'react';
import { useState } from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import { useDispatch } from 'react-redux'
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import { sendSocket } from 'actions/socket'


const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));



const FormCreateRoom = () => {
  const [login, setLogin] = useState('');
  const dispatch = useDispatch()
  const createRoom = () => {
    dispatch(sendSocket('server/create'))
  }
  const classes = useStyles();
  return (
    <div>
      <Container>
        <CssBaseline />
      <div className={classes.paper}>
        <Typography component="h1" variant="h5">
          Cr√©e une partie
        </Typography>
        <form className={classes.form} noValidate>
          <TextField
            value={login}
            onInput={e => setLogin(e.target.value)}
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="login"
            label="login"
            type="login"
            id="login"
            autoComplete="current-login"
          />
          <Button
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            onClick={createRoom}
            >
            Valider
          </Button>
        </form>
      </div>
      </Container>
    </div>
  );
}
export default FormCreateRoom
