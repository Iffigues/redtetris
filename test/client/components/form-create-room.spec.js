import React from "react";
import _ from 'lodash'
import '@testing-library/jest-dom/extend-expect';
import io from 'socket.io-client';
import Enzyme, { shallow } from "enzyme";
import { render, fireEvent } from '@testing-library/react'
import Adapter from "enzyme-adapter-react-16";
import { TestAppSocketProvider } from "../helpers/socketContext";
import FormCreateRoom from "../../../src/client/components/FormCreateRoom";

jest.mock('socket.io-client', () => {
  const mSocket = {
    emit: jest.fn(),
  };
  return jest.fn(() => mSocket);
});

Enzyme.configure({ adapter: new Adapter() });


describe('FormCreateRoom component', () => {

  const Wrapper = () => (
    <FormCreateRoom
      login="toto"
    />
  )
  
  it('Is exists', () => {
    const wrapper = shallow(<Wrapper />);
    expect(wrapper).toMatchSnapshot();
  });
  
  
  it('Can mount', () => {
    const Wr = () => (
      <TestAppSocketProvider>
        <FormCreateRoom
          login="toto"
        />
      </TestAppSocketProvider>
    )
    render(<Wr />);
  })

  
  it('No game', () => {
    const ENDPOINT = 'localhost:3004';
    const mockSocket = io(ENDPOINT);

    const Wr = () => (
      <TestAppSocketProvider>
        <FormCreateRoom
          login="toto"
        />
      </TestAppSocketProvider>
    )
    const { container } = render(<Wr />);
    const btn_create_room = container.querySelector('.test--btn-create-room')
    fireEvent.click(btn_create_room)
    expect(mockSocket.emit).toHaveBeenCalledTimes(1)
  })

});