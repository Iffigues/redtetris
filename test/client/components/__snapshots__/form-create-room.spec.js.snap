// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCreateRoom component Is exists 1`] = `
<FormCreateRoom
  login="toto"
/>
`;
